eva-accession-release/src/test/java/uk/ac/ebi/eva/accession/release/io/AccessionedVariantMongoReaderTest.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    @Test
+    public void linkedSubmittedVariants() throws Exception {
+        reader.open(executionContext);
+        Map<String, Variant> variants = new HashMap<>();
+        Variant variant;
+        while ((variant = reader.read()) != null) {
+            variants.put(variant.getMainId(), variant);
+        }
+        reader.close();
+        assertEquals(2, variants.size());
+        assertEquals(2, variants.get(RS_1).getSourceEntries().size());
+        assertEquals(1, variants.get(RS_2).getSourceEntries().size());
+    }
+
+    @Test
+    public void queryOtherAssembly() throws Exception {
+        reader = new AccessionedVariantMongoReader(ASSEMBLY_ACCESSION_2, mongoClient, TEST_DB);
+        reader.open(executionContext);
+        Map<String, Variant> variants = new HashMap<>();
+        Variant variant;
+        while ((variant = reader.read()) != null) {
+            variants.put(variant.getMainId(), variant);
+        }
+        assertEquals(1, variants.size());
+        assertEquals(2, variants.get(RS_3).getSourceEntries().size());
+    }
+
+    @Test
+    public void variantClassAttribute() throws Exception {
+        reader.open(executionContext);
+        Map<String, Variant> variants = new HashMap<>();
+        Variant variant;
+        while ((variant = reader.read()) != null) {
+            variants.put(variant.getMainId(), variant);
+        }
+        reader.close();
+        assertEquals(2, variants.size());
+        String snpSequenceOntology = "SO:0001483";
+        assertTrue(variants
+                           .get(RS_1)
+                           .getSourceEntries()
+                           .stream()
+                           .allMatch(se -> snpSequenceOntology.equals(se.getAttribute(VARIANT_CLASS_KEY))));
+        assertTrue(variants
+                           .get(RS_2)
+                           .getSourceEntries()
+                           .stream()
+                           .allMatch(se -> snpSequenceOntology.equals(se.getAttribute(VARIANT_CLASS_KEY))));
+    }
+
+    @Test
+    public void studyIdAttribute() throws Exception {
+        reader.open(executionContext);
+        Map<String, Variant> variants = new HashMap<>();
+        Variant variant;
+        while ((variant = reader.read()) != null) {
+            variants.put(variant.getMainId(), variant);
+        }
+        reader.close();
+        assertEquals(2, variants.size());
+
+        String studyId;
+        studyId = "PRJEB7923";
+        assertEquals(studyId, variants.get(RS_1).getSourceEntry(studyId, studyId).getAttribute(STUDY_ID_KEY));
+        studyId = "PRJEB9999";
+        assertEquals(studyId, variants.get(RS_1).getSourceEntry(studyId, studyId).getAttribute(STUDY_ID_KEY));
+        studyId = "PRJEB7923";
+        assertEquals(studyId, variants.get(RS_2).getSourceEntry(studyId, studyId).getAttribute(STUDY_ID_KEY));
+    }
+
+    @Test
+    public void clusteredVariantWithoutSubmittedVariants() throws Exception {
```

Comment:
A clustered variant without submitted variants should be allowed in a release?

Reply:
no, but skipping it in the reader would complicate the logic, it's easier to do a processor